File: client/src/main/java/net/treset/minecraftlauncher/util/ui/sort/LauncherModNameComparator.java
package net.treset.minecraftlauncher.util.ui.sort;

import net.treset.mc_version_loader.launcher.LauncherMod;
import net.treset.minecraftlauncher.LauncherApplication;

import java.util.Comparator;

public class LauncherModNameComparator implements Comparator<LauncherMod> {
    @Override
    public int compare(LauncherMod o1, LauncherMod o2) {
        return o1.getName().compareTo(o2.getName());
    }

    @Override
    public String toString() {
        if(LauncherApplication.stringLocalizer == null) {
            return "sort.name";
        }
        return LauncherApplication.stringLocalizer.get("sort.name");
    }
}

File: client/composeApp/src/commonMain/kotlin/net/treset/treelauncher/backend/util/sort/LauncherModNameComparator.kt
package net.treset.treelauncher.backend.util.sort

import net.treset.treelauncher.backend.data.LauncherMod
import net.treset.treelauncher.localization.strings

class LauncherModNameComparator : Comparator<LauncherMod> {
    override fun compare(o1: LauncherMod, o2: LauncherMod): Int {
        return o1.name.lowercase().compareTo(o2.name.lowercase())
    }

    override fun toString(): String = strings().sortBox.sort.name()
}
